spec/integration/an_organized_organizer_with_after_callbacks_deferred.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe 'An organized organizer with after callbacks deferred', type: :integration do
+  let!(:test_interactor_1) do
+    build_interactor('TestInteractor1') do
+      defer_after_callbacks_when_organized
+
+      after_perform do
+        context.after_perform_1a = context.after_perform_1b + 1
+      end
+
+      after_perform do
+        context.after_perform_1b = context.after_perform_2 + 1 
+      end
+
+      def perform
+        context.perform_1 = 1
+      end
+    end
+  end
+
+  let!(:test_interactor_2) do
+    build_interactor('TestInteractor2') do
+      after_perform do
+        context.after_perform_2 = context.perform_2 + 1
+      end
+
+      def perform
+        context.perform_2 = context.after_perform_3a + 1
+      end
+    end
+  end
+
+  let!(:test_interactor_3) do
+    build_interactor('TestInteractor3') do
+      defer_after_callbacks_when_organized
+
+      after_perform do
+        context.after_perform_3a = context.after_perform_3b + 1
+      end
+
+      after_perform do
+        context.after_perform_3b = context.after_perform_4a + 1
+      end
+
+      def perform
+        context.perform_3 = context.perform_1 + 1
+      end
+    end
+  end
+
+  let!(:test_interactor_4) do
+    build_interactor('TestInteractor4') do
+      after_perform do
+        context.after_perform_4a = context.after_perform_4b + 1
+      end
+
+      after_perform do
+        context.after_perform_4b = context.perform_4 + 1
+      end
+
+      def perform
+        context.perform_4 = context.perform_3 + 1
+      end
+    end
+  end
+
+  let!(:test_sub_organizer_5) do
+    build_organizer('TestSubOrganizer5') do
+      defer_after_callbacks_when_organized
+
+      after_perform do
+        context.after_perform_5a = context.after_perform_5b + 1
+      end
+
+      after_perform do
+        context.after_perform_5b = context.after_perform_1a + 1
+      end
+
+      organize TestInteractor3, TestInteractor4
+    end
+  end
+
+  let(:interactor_class) do
```

Comment:
Technically an Organizer is both an `interactor` _and_ and `organizer` since the latter inherits from the former (you can organize Organizers). 

I actually tried to update that to `organizer_class` though just to be a little clearer, but the RSpec Shared Examples are expecting `interactor_class` (which isn't _entirely_ wrong since an Organizer is an Interactor too)

Reply:
🔥 thanks for digging!